test/e2e/tls/tls.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2019 Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package tls
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	pmempod "github.com/intel/pmem-csi/test/e2e/pod"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/format"
+)
+
+var _ = Describe("TLS", func() {
+	f := framework.NewDefaultFramework("tls")
+
+	// All of the following pod names, namespaces and ports match
+	// those in the current deployment files.
+
+	var nodePod *v1.Pod
+	BeforeEach(func() {
+		// Find one node driver pod.
+		label := labels.SelectorFromSet(labels.Set(map[string]string{"app": "pmem-csi-node"}))
+		pods, err := f.ClientSet.CoreV1().Pods("default").List(metav1.ListOptions{LabelSelector: label.String()})
+		framework.ExpectNoError(err, "list PMEM-CSI node pods")
+		Expect(pods.Items).NotTo(BeEmpty(), "have PMEM-CSI node pods")
+		nodePod = &pods.Items[0]
+	})
+
+	Context("controller", func() {
+		It("is secure", func() {
+			checkTLS(f, "pmem-csi-controller-0.pmem-csi-controller.default")
```

Comment:
Could we avoid this hardcoding name, as it depends on the driver's namespace.

Reply:
I could look for a running PMEM-CSI driver, but that leads to a bigger question: which one should it use if there is more than one?

How do I test that the code that does the lookup work?

I prefer to solve that together with actually installing the different in different ways, including installing it in different namespaces. I want to work on that this week.

Reply:
> I could look for a running PMEM-CSI driver, but that leads to a bigger question: which one should it use if there is more than one?

Could you consider using the controller's service ip.

Reply:
@pohly otherwise, we can deal with this part of operator changes as it also changes the name of the controller service.